### PR TITLE
Lps 171717 | Add tests for Object REST API with different scopes in an LXC environment

### PR DESCRIPTION
**Pull request by yanan**
**Background:**
- set include-and-override=portal-liferay-online.properties in portal-ext.properties
- Given an OAuth application with headless scopes is created on default instance
- And Given a new virtual instances www.able.com is created
- And Given an OAuth application with headless scopes is created on www.able.com
- And Given objectDefinition University with site scope and new field name is created and published in default instance
- And Given objectDefinition University with default company scope and new field location is created and published in www.able.com

**Test Plan:**
- When I use postScopeScopeKey and scopeKey to create an object entry in default instance
Then the object entry created successfully 
- When I use postTest to create an object entry in www.able.com instance
Then the object entry with location created successfully
- When navigate to www.able.com:8080/o/c/tests
Then location field exists and name doesn't exist in postTest body
And Then location field exists and name doesn't exist in Test schema

**Ticket:**
- https://issues.liferay.com/browse/LPS-171717

### CODE REVIEW

**Comment by yanan** [comment]
ci:test:sf

**Comment by yanan** [comment]
ci:test:headless-functional:nocompile

**Comment by yanan** [comment]
ci:test:oauth2

**Review by Hijky — APPROVED** [review]

**Comment by yanan** [comment]
Test failures are unrelated.
- Here is the Oauth [tests result on Testray](https://testray.liferay.com/home/-/testray/runs?testrayBuildId=2219495009). And there is [no unique oauth test failures](https://testray.liferay.com/home/-/testray/runs/compare?formDate=1677837926249&testrayRunIdA=2219495070&testrayRunIdB=2219381275) related to the pr.

**Comment by yanan** [comment]
ci:test:headless-functional:nocompile

**Comment by yanan** [comment]
ci:test:headless-functional:nocompile

File: modules/apps/object/object-rest-test/src/testFunctional/macros/OAuth2ForObject.macro [hunk 1]
```diff
@@ -0,0 +1,55 @@
+definition {
+
+	macro addAuth2Application {
```
Comment (magjed4289) [inline modules/apps/object/object-rest-test/src/testFunctional/macros/OAuth2ForObject.macro:3]: Can we change the name to addOAuth2Application? This way we stay aligned with OAuth2 we use after :)

File: modules/apps/object/object-rest-test/src/testFunctional/macros/OAuth2ForObject.macro [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+		var accessToken = OAuth2.getAccessToken(
+			client_id = "myApplication",
+			client_secret = "mySecret",
+			grantFlow = "clientCredentials",
+			portalURL = ${baseURL});
+
+		return ${accessToken};
+	}
+
+	macro setUpGlobalTokenForDefaultInstance {
```
Comment (magjed4289) [inline modules/apps/object/object-rest-test/src/testFunctional/macros/OAuth2ForObject.macro:45]: could we divide it here in two methods? first one to create the token and then another one to make it static? The setUpGlobalTokenForDefaultInstance could be called staticDefaultInstanceToken and have the defaultInstanceToken as a parameter

File: modules/apps/object/object-rest-test/src/testFunctional/macros/ObjectDefinitionAPI.macro [hunk 3]
```diff
@@ -290,11 +285,42 @@ definition {
 				-u test@liferay.com:test
 		''';
 
+<<<<<<< HEAD
```
Comment (magjed4289) [inline modules/apps/object/object-rest-test/src/testFunctional/macros/ObjectDefinitionAPI.macro:288]: A conflict is not solver here

**Review by magjed4289 — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-rest-test/src/testFunctional/macros/OAuth2ForObject.macro (hunk 1 above)
Comment (yanan) [reply modules/apps/object/object-rest-test/src/testFunctional/macros/OAuth2ForObject.macro]: Updated, thanks

File: modules/apps/object/object-rest-test/src/testFunctional/macros/ObjectDefinitionAPI.macro (hunk 3 above)
Comment (yanan) [reply modules/apps/object/object-rest-test/src/testFunctional/macros/ObjectDefinitionAPI.macro]: Oops, updated :heart:

File: modules/apps/object/object-rest-test/src/testFunctional/macros/OAuth2ForObject.macro (hunk 2 above)
Comment (yanan) [reply modules/apps/object/object-rest-test/src/testFunctional/macros/OAuth2ForObject.macro]: I've removed the setUpGlobalTokenForDefaultInstance() which is not called anymore. There is a method createTokenWithOAuth2Scopes to create token which called in testcase file.

**Review by magjed4289 — APPROVED** [review]

**Comment by yanan** [comment]
ci:forward

**Pull request closed without merging**